### PR TITLE
boot_nic_with_iommu:Add one case to support viommu device

### PR DESCRIPTION
**Pull request by yanglei-rh**
Boot nic with iommu_platform=on based on virtio iommu device

ID:2148335
Signed-off-by: Lei Yang <leiayang@redhat.com>

### CODE REVIEW

**Comment by yanglei-rh** [comment]
Test pass:
 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.boot_guest_with_viommu.direct_plug.q35: PASS (86.30 s)
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.boot_guest_with_viommu.non_direct_plug.q35: PASS (70.23 s)

**Comment by yanglei-rh** [comment]
Hello @PaulYuuu Could you please help me review this patch? Thanks a lot.

File: qemu/tests/cfg/boot_guest_with_viommu.cfg [hunk 1]
```diff
@@ -0,0 +1,16 @@
+- boot_guest_with_viommu:
```
Comment (PaulYuuu) [inline qemu/tests/cfg/boot_guest_with_viommu.cfg:1]: Suggest moving to `boot_nic_with_iommu`

File: qemu/tests/cfg/boot_guest_with_viommu.cfg [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    only Linux
+    only virtio_net
+    serial_login = no
+    login_timeout = 240
+    virtio_iommu = yes
+    nic_extra_params = ",iommu_platform=on"
+    variants:
+        - direct_plug:
+              virtio_iommu_direct_plug = yes
+        - non_direct_plug:
```
Comment (PaulYuuu) [inline qemu/tests/cfg/boot_guest_with_viommu.cfg:15]: This is not a useful scenario.

Comment (yanglei-rh) [reply qemu/tests/cfg/boot_guest_with_viommu.cfg:15]: Ok, thanks for sharing this info. It will be changed in the next update.

**Comment by yanglei-rh** [comment]
Test Pass:
 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.intel_iommu.q35: PASS (155.01 s)
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.virtio_iommu.q35: PASS (82.76 s)

**Comment by yanglei-rh** [comment]
@PaulYuuu Please help me review it again, thanks a lot.

File: qemu/tests/cfg/boot_nic_with_iommu.cfg [hunk 3]
```diff
@@ -1,23 +1,36 @@
...
-    vhostforce = on
+    variants:
+        - intel_iommu:
+            x86_64, i386:
+                only q35
+            no WinXP WinVista Win7 Win8 Win8.1 Win2003
+            no Win2008 Win2008..r2 Win2012 Win2012..r2
+            machine_type_extra_params = "kernel-irqchip=split"
+            HostCpuVendor.intel:
+                intel_iommu = yes
```
Comment (PaulYuuu) [inline qemu/tests/cfg/boot_nic_with_iommu.cfg:16]:
```suggestion
            only HostCpuVendor.intel
            intel_iommu = yes
```

File: qemu/tests/cfg/boot_nic_with_iommu.cfg [hunk 4]
```diff
@@ -1,23 +1,36 @@
...
+                iommu_caching_mode = on
+            Linux:
+                enable_guest_iommu = yes
+            nic_extra_params = ",disable-legacy=on,disable-modern=off,iommu_platform=on,ats=on"
+            aarch64:
+                machine_type_extra_params += ",iommu=smmuv3"
+                nic_extra_params = ",iommu_platform=on,ats=on"
+            s390x:
+                nic_extra_params = ",iommu_platform=on"
+                vhostforce = on
```
Comment (PaulYuuu) [inline qemu/tests/cfg/boot_nic_with_iommu.cfg:26]: Intel iommu is only for x86, can consider letting them separate.

File: qemu/tests/cfg/boot_nic_with_iommu.cfg [hunk 5]
```diff
@@ -1,23 +1,36 @@
...
+        - virtio_iommu:
+            virt_test_type = qemu
```
Comment (PaulYuuu) [inline qemu/tests/cfg/boot_nic_with_iommu.cfg:28]: Dup.

File: qemu/tests/cfg/boot_nic_with_iommu.cfg [hunk 6]
```diff
@@ -1,23 +1,35 @@
...
     type = boot_nic_with_iommu
-    nic_extra_params = ",disable-legacy=on,disable-modern=off,iommu_platform=on,ats=on"
-    aarch64:
-        machine_type_extra_params += ",iommu=smmuv3"
-        nic_extra_params = ",iommu_platform=on,ats=on"
-    s390x:
-        nic_extra_params = ",iommu_platform=on"
-    vhostforce = on
+    variants:
+        - intel_iommu:
```
Comment (PaulYuuu) [inline qemu/tests/cfg/boot_nic_with_iommu.cfg:9]: The name is not suitable for non-intel CPU and other architectures.

**Comment by yanglei-rh** [comment]
Test Pass:
(1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.iommu.q35: STARTED
 (1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.iommu.q35: PASS (122.92 s)
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.virtio_iommu.q35: STARTED
 (2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.virtio_iommu.q35: PASS (71.69 s)

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**